pkg/kubelet/lifecycle/predicate_test.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package lifecycle
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/scheduler/schedulercache"
+)
+
+var (
+	quantity              = *resource.NewQuantity(1, resource.DecimalSI)
+	extendedResourceName1 = "example.com/er1"
+	extendedResourceName2 = "example.com/er2"
+)
+
+func TestRemoveMissingExtendedResources(t *testing.T) {
+	for _, test := range []struct {
+		pod         *v1.Pod
+		node        *v1.Node
+		expectedPod *v1.Pod
+	}{
+		{
+			pod: makeTestPod(
+				v1.ResourceList{},
+				v1.ResourceList{"foo.com/bar": quantity},
+			),
+			node: makeTestNode(
+				v1.ResourceList{"foo.com/baz": quantity},
+			),
+			expectedPod: makeTestPod(
+				v1.ResourceList{},
+				v1.ResourceList{"foo.com/bar": quantity},
```

Comment:
Hmm, shouldn't this resource request be removed?

Reply:
The request is in `Limits` which is skipped in this logic. I will refine this test once the overall design is LGTM'd.

Reply:
Oh I see. So this is limit and we leave limit unchanged. For extended resources, we actually require limits to equal to requests and in devicemanager, we actually examine resource Limits during resource allocation. Can we make sure to also remove limits for non-existing extended resources?

Reply:
Never mind. Whether we have Limit or not doesn't affect GeneralPredicate evaluation.